clar2wasm/src/tools.rs

```diff
@@ -376,6 +378,46 @@ struct CrossEvalResult {
     compiled: Result<Option<Value>, Error>,
 }
 
+#[derive(Debug, Clone, Copy)]
+enum KnownBug {
+    /// [https://github.com/stacks-network/stacks-core/issues/4622]
+    ListOfQualifiedPrincipal,
+}
+
+impl KnownBug {
+    fn check_for_knonw_bugs(
+        compiled: &Result<Option<Value>, Error>,
+        interpreted: &Result<Option<Value>, Error>,
+    ) -> Option<Self> {
+        let check_predicate = |pred: &dyn Fn(&Error) -> bool| {
+            interpreted.as_ref().is_err_and(pred) && compiled.as_ref().is_err_and(pred)
+        };
+
+        if check_predicate(&Self::has_list_of_qualified_principal_issue) {
+            Some(KnownBug::ListOfQualifiedPrincipal)
+        } else {
+            None
+        }
+    }
+
+    /// Allows to detect if an error suffers from this issue:
+    /// [https://github.com/stacks-network/stacks-core/issues/4622].
+    fn has_list_of_qualified_principal_issue(err: &Error) -> bool {
+        static RGX: LazyLock<Regex> = LazyLock::new(|| {
+            let regex = r#"expecting expression of type '\(principal ([^\)]+)\)', found '\(principal ([^\)]+)\)'"#;
```

Comment:
Should [this error](https://github.com/stacks-network/clarity-wasm/actions/runs/10642286761/job/29504515495#step:4:4057) also be caught by the regex?

Reply:
Lol at this, I have a test with `index-of` that provokes an error that matches the regex, and here is a test with the same `index-of` that spawns a different message :D